tests/integration_test.py

```diff
@@ -101,6 +103,13 @@ def tearDown(self):
                 pass
         for folder in self.tmp_folders:
             shutil.rmtree(folder)
+
+        for volume in self.tmp_volumes:
+            try:
+                self.client.remove_volume(volume)
+            except docker.errors.APIError:
+                pass
```

Comment:
I don't think the improvement is in performance, it's in clarify.

With a single base class for all tests, it's not clear which tests actually make use of this feature.  Refactoring and moving the test suite around becomes harder because everything needs a single base.

With a yield fixture it's explicit that the test requires volume_cleanup. 

Not sure how it would make the code more complex.